typescript/src/generator.ts

```diff
@@ -181,15 +181,17 @@ export class Generator {
   }
 
   private buildAPIObject(): API {
-    const protoFilesToGenerate = this.request.protoFile.filter(
-      pf =>
-        pf.name &&
-        this.request.fileToGenerate.includes(pf.name) &&
-        !API.isIgnoredService(pf)
-    );
-    const packageNamesToGenerate = protoFilesToGenerate.map(
-      pf => pf.package || ''
-    );
+    const protoPackagesToGenerate = new Set<string>();
+    API.filterOutIgnoredServices(
+      this.request.protoFile.filter(
+        fd =>
+          this.request.fileToGenerate.includes(fd.name!) &&
+          fd.service &&
+          fd.service.length > 0
+      )
+    ).forEach(fd => protoPackagesToGenerate.add(fd.package || ''));
+    console.warn(protoPackagesToGenerate);
```

Comment:
Do you want to leave this debug message in?

Reply:
No! Thank you for spotting it.